conda/gateways/connection/adapters/jlap/repodata_proxy.py

```diff
@@ -243,9 +250,15 @@ def send(request: requests.PreparedRequest, base_adapter):
     return response
 
 
+class FileResponse(NamedTuple):
```

Comment:
We've talked about this a lot, Pydantic or attrs would be good to use but would add a non-trivial dependency, and dataclasses have a performance impact that it may not be worth relying on. NamedTuple OTOH is known to be considered a legacy API by the core devs. Not a lot of good options I'm afraid.

Reply:
I think I like it. We get a constructor for free.

Reply:
I don't think this feature is going to be deprecated (at least I hope not!): https://docs.python.org/3/library/typing.html?highlight=typing#typing.NamedTuple

I've always assumed that `NamedTuple` more or less equal to writing the equivalent class:

```python
class Something:
    __slots__ = ('field_one', 'field_two')
    
    def __init__(self, field_one, field_two):
        self.field_one = field_one
        self.field_two = field_two
```

NamedTuples are also "frozen" so you are unable to modify them after creation. I use them a lot when building immutable objects.